app/forms/floating-ip-create.tsx

```diff
@@ -0,0 +1,158 @@
+   * file. The callers that don't need the arg can ignore it.
+   */
+  onDismiss?: (navigate: NavigateFunction) => void
+  onSuccess?: (floatingIp: FloatingIp) => void
+}
+
+export function CreateFloatingIpSideModalForm({
+  onSubmit,
+  onSuccess,
+}: CreateSideModalFormProps) {
+  // Fetch 1000 to we can be sure to get them all. There should only be a few
+  // anyway. Not prefetched because the prefetched one only gets 25 to match the
+  // query table. This req is better to do async because they can't click make
+  // default that fast anyway.
+  const { data: allPools } = useApiQuery('projectIpPoolList', {
+    query: { limit: 1000 },
+  })
+
+  const defaultPool = useMemo(
+    () => (allPools ? allPools.items.find((p) => p.isDefault)?.name : undefined),
+    [allPools]
+  )
+
+  const defaultValues: FloatingIpCreate = {
+    name: '',
+    description: '',
+    // defaultPool doesn't seem to be getting set in the form when page is loaded directly
+    pool: defaultPool,
+    address: undefined,
+  }
+
+  const queryClient = useApiQueryClient()
+  const projectSelector = useProjectSelector()
+  const addToast = useToast()
+  const navigate = useNavigate()
+
+  const createFloatingIp = useApiMutation('floatingIpCreate', {
+    onSuccess(data) {
+      queryClient.invalidateQueries('floatingIpList')
+      addToast({ content: 'Your Floating IP has been created' })
+      onSuccess?.(data)
+      navigate(pb.floatingIps(projectSelector))
+    },
+  })
+
+  const form = useForm({ defaultValues })
+
+  const toListboxItem = (p: SiloIpPool) => {
+    if (p.name !== defaultPool) {
+      return {
+        value: p.name,
+        label: p.name,
+      }
+    }
+    // For the default pool, add a label to the dropdown
+    return {
+      value: p.name,
+      labelString: p.name,
+      label: (
+        <>
+          {p.name}{' '}
+          <span className="text-quaternary selected:text-accent-secondary">(default)</span>
+        </>
+      ),
+    }
+  }
+
+  const [poolName] = useWatch({ control: form.control, name: ['pool'] })
+  const isPoolSelected = poolName && poolName.length > 0
+
+  return (
+    <SideModalForm
+      id="create-floating-ip-form"
+      title="Create Floating IP"
+      form={form}
+      onDismiss={() => navigate(pb.floatingIps(projectSelector))}
+      onSubmit={({ ...rest }) => {
+        const body = { ...rest }
+        onSubmit
+          ? onSubmit(body)
+          : createFloatingIp.mutate({ query: projectSelector, body })
+      }}
+      loading={createFloatingIp.isPending}
+      submitError={createFloatingIp.error}
+    >
+      <NameField name="name" control={form.control} />
+      <DescriptionField name="description" control={form.control} />
+      {allPools && (
+        <ListboxField
+          name="pool"
+          items={allPools?.items.map((p) => toListboxItem(p))}
+          label="Pool"
+          required
+          control={form.control}
+        />
+      )}
```

Comment:
This field not be required. Also by using `usePrefetchedApiQuery` above you can get rid of the `allPools` check because you know it's defined.